src/flattenizer.js

```diff
@@ -92,23 +85,23 @@ export const unflatten = (flattened, delimiter = '.') => {
  * @private
  */
 const explodeProperty = (currUnflattened, key, flattenedObj, delimiter) => {
-    const keys = key.split(delimiter);
-    const value = flattenedObj[key];
-    const lastKeyIndex = keys.length - 1;
+  const keys = key.split(delimiter);
+  const value = flattenedObj[key];
+  const lastKeyIndex = keys.length - 1;
 
-    for (let idx = 0; idx < lastKeyIndex; idx++) {
-        const currKey = keys[idx];
-        let nextKeyVal;
+  for (let idx = 0; idx < lastKeyIndex; idx++) {
+    const currKey = keys[idx];
+    let nextKeyVal;
 
-        if (!(currKey in currUnflattened)) {
-            nextKeyVal = parseInt(keys[idx + 1], 10);
-            currUnflattened[currKey] = isNaN(nextKeyVal) ? {} : [];
-        }
-
-        currUnflattened = currUnflattened[currKey];
+    if (!currUnflattened.hasOwnProperty(currKey)) {
+      nextKeyVal = parseInt(keys[idx + 1], 10);
+      currUnflattened[currKey] = isNaN(nextKeyVal) ? {} : [];
     }
 
-    currUnflattened[keys[lastKeyIndex]] = value;
+    currUnflattened = currUnflattened[currKey];
+  }
+
+  currUnflattened[keys[lastKeyIndex]] = value;
```

Comment:
Double nested arrays are pretty hard to read. Please pull in React and use JSX to solve this readability issue.